### PR TITLE
Poc replace body subset

### PR DESCRIPTION
**Pull request by ahaurw01**
Proof of concept:

If the old page and new page both have a `[data-turbolinks-root]` element, only replace that instead of the whole `body` element.

This is to test a hypothesis that replacing the whole body is what leads to our ad viewability not working properly.

cc @danfastspanishdict @mdzeng @nanek

### CODE REVIEW

File: src/turbolinks/snapshot_renderer.coffee [hunk 1]
```diff
@@ -70,7 +71,12 @@ class Turbolinks.SnapshotRenderer extends Turbolinks.Renderer
       replaceElementWithElement(inertScriptElement, activatedScriptElement)
 
   assignNewBody: ->
-    document.body = @newBody
+    existingRootElement = document.querySelector('[data-turbolinks-root]')
+    if @newRootElement and existingRootElement
+      console.log 'replacing root element only'
+      replaceElementWithElement existingRootElement, @newRootElement
```
Comment (seankarson) [inline src/turbolinks/snapshot_renderer.coffee]: @ahaurw01 Will this have the effect of ignoring the `data-turbolinks-permanent` tag that we have on our ad units? I think we saw a revenue boost from keeping those divs around a while back. Is that right, @nanek ?

File: src/turbolinks/snapshot_renderer.coffee [hunk 2]
```diff
@@ -50,21 +52,34 @@ class Turbolinks.SnapshotRenderer extends Turbolinks.Renderer
...
+  activateNewBodyScriptElements: ->
+    for inertScriptElement in @getNewBodyScriptElements()
+      activatedScriptElement = @createScriptElement(inertScriptElement)
+      replaceElementWithElement(inertScriptElement, activatedScriptElement)
 
   assignNewBody: ->
-    document.body = @newBody
+    existingRootElement = document.querySelector('[data-turbolinks-root]')
+    if @newRootElement and existingRootElement
+      console.log 'replacing root element only'
```
Comment (seankarson) [inline src/turbolinks/snapshot_renderer.coffee]: do we want this console log for our test?

**Comment by ahaurw01** [comment]
Thanks for the good comments here. This PR is not intended to merge, but is for POC purposes. I will create an official PR meant to merge shortly.

**Comment by ahaurw01** [comment]
Closing in favor of #2

**Pull request closed without merging**